app/src/main/java/com/example/reflector_android/Models/BlogRecyclerAdapter.kt

```diff
@@ -1,53 +1,66 @@
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.annotation.RequiresApi
 import androidx.recyclerview.widget.RecyclerView
 import com.example.reflector_android.R
 import com.example.reflector_android.network.Article
 import kotlinx.android.synthetic.main.article_item.view.*
 
-class BlogRecyclerAdapter(val articles: MutableList<Article>?) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+class BlogRecyclerAdapter(val articles: MutableList<Article>?) : RecyclerView.Adapter<CustomViewHolder>() {
     companion object {
         var isLoading: Boolean = false
     }
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomViewHolder {
 
         val layoutInflater = LayoutInflater.from(parent.context)
         val cellForRow = layoutInflater.inflate(R.layout.article_item, parent, false)
 
         return CustomViewHolder(cellForRow)
     }
 
-    @RequiresApi(Build.VERSION_CODES.O)
-    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
 
+    override fun getItemCount(): Int {
+        return articles?.size ?: 0
+    }
+
+    override fun onBindViewHolder(holder: CustomViewHolder, position: Int) {
         val article = articles?.get(position)
 
         when (article?.author) {
             null -> {
-                holder.itemView.progressBar.visibility = View.VISIBLE
-                holder.itemView.description.text = ""
-                holder.itemView.title.text = ""
-                holder.itemView.author.text = ""
-                holder.itemView.pubDate.text = ""
+                holder.customView.progressBar.visibility = View.VISIBLE
+                holder.customView.description.text = ""
+                holder.customView.title.text = ""
+                holder.customView.author.text = ""
+                holder.customView.pubDate.text = ""
             }
             else -> {
-                holder.itemView.progressBar.visibility = View.GONE
-                holder.itemView.description.text = article.description
-                holder.itemView.title.text = article.title
-                holder.itemView.author.text = article.author
-                holder.itemView.pubDate.text = article.pubDate.toString()
+                holder.customView.progressBar.visibility = View.GONE
+                holder.customView.description.text = article.description
+                holder.customView.title.text = article.title
+                holder.customView.author.text = article.author
+                holder.customView.pubDate.text = article.pubDate.toString()
+                holder.url = article
             }
         }
     }
+}
 
-    override fun getItemCount(): Int {
-        return articles?.size?: 0
-    }
+class CustomViewHolder(val customView: View, var url: Article? = null) : RecyclerView.ViewHolder(customView) {
```

Comment:
1. If the CustomViewHolder is the article cell, we should rename the class as such or to something more specific to its purpose. This is because you may have to create more ViewHolder subclasses in the future that are very different from each other and CustomViewHolder can't distinguish well between them all.

2. In my opinion, we should move this class to its own separate file and also make that file outside of the "Models" folder.